hooks/useClickOutside.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect } from 'react';
+import type { RefObject } from 'react';
+
+// eslint-disable-next-line no-unused-vars
+type TypeHandler = (event: MouseEvent | TouchEvent) => void;
+
+export const useClickOutside = (
```

Comment:
Can you outsource the type of this Hook?

`type UseClickOutsideProps = { ref: RefObject<HTMLElement>, handler: TypeHandler }`

`export const useClickOutside = (props: UseClickOutsideProps) => ...`

Reply:
Sure!